staging/src/k8s.io/cloud-provider/controllers/node/node_controller.go

```diff
@@ -238,7 +238,12 @@ func (cnc *CloudNodeController) syncHandler(key string) error {
 		return err
 	}
 
-	err = cnc.syncNode(context.TODO(), node)
+	cloudTaint := getCloudTaint(node.Spec.Taints)
+	if cloudTaint == nil {
```

Comment:
@andrewsykim can we instead check cloudTaint here?
I think we still need to keep getting the node resource on Line 230 and use that in Line 251

Reply:
Would prefer to remove it here actually since there's an additional cache read here we don't need. For the event recorder on line 251, can we call that near the end of `syncNode` instead?

Reply:
gotcha, let me update that